src/EntityComponentManager.cc

```diff
@@ -1367,6 +1393,7 @@ void EntityComponentManager::SetState(
           }
         }
         this->SetChanged(entity, compIter.first, flag);
+        this->dataPtr->AddModifiedComponent(entity);
```

Comment:
I think the `SetChanged` call above will also call `AddModifiedComponent`

Reply:
Oops, thanks 70c432050ff7c18d5cb3f79de5cb697ce950ff14